storage/tx.go

```diff
@@ -0,0 +1,61 @@
+package storage
+
+type PendingTxType []byte
+
+// StateTransitionTx indicates a pending state transition transaction
+var StateTransitionTx PendingTxType = []byte("st/")
+
+// pendingTxs method retrieves the list of pending transaction hashes for a
+// given process ID. If no pending transactions are found, it returns an empty
+// list. If an error occurs during retrieval, it returns the error. This
+// method is used internally to manage and track pending transactions
+// associated and should be called with appropriate locking to ensure thread
+// safety.
+func (s *Storage) pendingTxs(txType PendingTxType, processID []byte) (bool, error) {
+	// Get current pending txs
+	var pendingTx bool
+	prefix := append(pendingTxPrefix, txType...)
+	if err := s.getArtifact(prefix, processID, &pendingTx); err != nil {
+		if err != ErrNotFound {
+			return false, err
+		}
+		// If not found, start with empty list
+		return false, nil
+	}
+	return pendingTx, nil
+}
+
+// SetPendingTx marks a process as having a pending on-chain
+// transaction. If the process already has a pending transaction, it does
+// nothing. If an error occurs during the operation, it returns the error.
+func (s *Storage) SetPendingTx(txType PendingTxType, processID []byte) error {
+	s.globalLock.Lock()
```

Comment:
Do we actually need to global Lock here? Just asking to be sure we are properly using the lock.